ldap_reader/reader.py

```diff
@@ -115,6 +117,92 @@ def _determine_group_type(self, ldap_id):
 
         return 'group'
 
+    def can_auth(self, username, password):
+        '''
+        Checks the ability of the given username and
+        password to connect to the AD.
+        Returns True if valid, false if not.
+        '''
+        log = logging.getLogger("can_auth")
+        # Throw out empty passwords.
+        if password == "":
+            return False
+
+        # Use another ldap connection to test user authentication
```

Comment:
So this is why I left it as a module-level function instead of part of the objects originally. In addition, the way netkes uses this function, it expects to call a `<module>.can_auth(username, password)` function. I'm happy to give @bdzim more work to do here in the eventual refactor to fold this on (just duck type connections for auth), but it's something to keep in mind that I probably didn't tell you about beforehand.